topostats/tracing/disordered_tracing.py

```diff
@@ -0,0 +1,708 @@
+    mask_smoothing_params : dict
+        Dictionary of parameters to smooth the grain mask for better quality skeletonisation results. Contains
+        a gaussian 'sigma' and number of dilation iterations.
+    skeletonisation_params : dict
+        Dictionary of options for skeletonisation, options are 'zhang' (scikit-image) / 'lee' (scikit-image) / 'thin'
+        (scikitimage) or 'topostats' (original TopoStats method).
+    pruning_params : dict
+        Dictionary of options for pruning.
+    pad_width : int
+        Padding to the cropped image mask.
+    cores : int
+        Number of cores to process with.
+
+    Returns
+    -------
+    tuple[dict, dict]
+        Binary and interger labeled cropped and full-image masks from skeletonising and pruning the grains in the image.
+    """
+    # Check both arrays are the same shape - should this be a test instead, why should this ever occur?
+    if image.shape != grains_mask.shape:
+        raise ValueError(f"Image shape ({image.shape}) and Mask shape ({grains_mask.shape}) should match.")
+
+    cropped_images, cropped_masks, bboxs = prep_arrays(image, grains_mask, pad_width)
+    n_grains = len(cropped_images)
+    img_base = np.zeros_like(image)
+    disordered_trace_crop_data = {}
+
+    # want to get each cropped image, use some anchor coords to match them onto the image,
+    #   and compile all the grain images onto a single image
+    all_images = {
+        "smoothed_grain": img_base.copy(),
+        "skeleton": img_base.copy(),
+        "pruned_skeleton": img_base.copy(),
+    }
+
+    LOGGER.info(f"[{filename}] : Calculating DNA tracing statistics for {n_grains} grains.")
+
+    for cropped_image_index, cropped_image in cropped_images.items():
+        cropped_mask = cropped_masks[cropped_image_index]
+        disordered_trace_images = trace_grain(
+            cropped_image=cropped_image,
+            cropped_mask=cropped_mask,
+            pixel_to_nm_scaling=pixel_to_nm_scaling,
+            mask_smoothing_params=mask_smoothing_params,
+            skeletonisation_params=skeletonisation_params,
+            pruning_params=pruning_params,
+            filename=filename,
+            min_skeleton_size=min_skeleton_size,
+            n_grain=cropped_image_index,
+        )
+        LOGGER.info(f"[{filename}] : Disordered Traced grain {cropped_image_index + 1} of {n_grains}")
+
+        # remap the cropped images back onto the original
+        for image_name, full_image in all_images.items():
+            crop = disordered_trace_images[image_name]
+            bbox = bboxs[cropped_image_index]
+            full_image[bbox[0] : bbox[2], bbox[1] : bbox[3]] += crop[pad_width:-pad_width, pad_width:-pad_width]
+
+        disordered_trace_crop_data[f"grain_{cropped_image_index}"] = disordered_trace_images
+        disordered_trace_crop_data[f"grain_{cropped_image_index}"]["bbox"] = bboxs[cropped_image_index]
+
+    return disordered_trace_crop_data, all_images
+
+
+# not used
```

Comment:
Is there a need to keep unused functions? They will add a burden to writing of tests so unless there is a need to keep them around I'd suggest removing them.

Reply:
Removed. I think it was an old diagnostic image when things wern't working and anyway it should be in splining. Good find!